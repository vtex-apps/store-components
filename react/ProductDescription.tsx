import React, { memo } from 'react'
import type { MemoExoticComponent, PropsWithChildren } from 'react'
import { FormattedMessage, useIntl } from 'react-intl'
import { useCssHandles } from 'vtex.css-handles'
import type { CssHandlesTypes } from 'vtex.css-handles'
import { formatIOMessage } from 'vtex.native-types'
import { useProduct } from 'vtex.product-context'

import { SanitizedHTML, DEFAULTS } from './components/SanitizedHTML'
import GradientCollapse from './components/GradientCollapse'

const CSS_HANDLES = [
  'productDescriptionContainer',
  'productDescriptionTitle',
  'productDescriptionText',
] as const

type Props = {
  /** Description fallback */
  description?: string
  /** Section title */
  title?: string
  /** Define whether or not to show the title */
  showTitle?: boolean
  /** Define if content should start collapsed or not */
  collapseContent?: boolean
  /** Used to override default CSS handles */
  classes?: CssHandlesTypes.CustomClasses<typeof CSS_HANDLES>
}

const allowedTags = [
  ...DEFAULTS.allowedTags,
  'link',
  'body',
  'html',
  'style',
  'link',
  'script',
  'head',
  'meta',
  'object',
  'embed',
]

const allowedAttributes = {
  ...DEFAULTS.allowedAttributes,
  meta: ['charset', 'name', 'content'],
  object: ['type', 'height', 'width', 'data'],
  embed: ['height', 'width', 'src'],
}

/**
 * Product Description Component.
 * Render the description of a product
 */
function ProductDescription(props: PropsWithChildren<Props>) {
  const { handles } = useCssHandles(CSS_HANDLES, { classes: props.classes })
  const intl = useIntl()
  const { product } = useProduct()

  const description = props.description ?? product?.description

  if (!description) {
    return null
  }

  const { collapseContent = true, showTitle = true, title } = props

  return (
    <div className={handles.productDescriptionContainer}>
      {showTitle && (
        <FormattedMessage id="store/product-description.title">
          {txt => (
            <h2
              className={`${handles.productDescriptionTitle} t-heading-5 mb5 mt0`}
            >
              {title ? formatIOMessage({ id: title, intl }) : txt}
            </h2>
          )}
        </FormattedMessage>
      )}

      <div className={`${handles.productDescriptionText} c-muted-1`}>
        {collapseContent ? (
          <GradientCollapse collapseHeight={220} product={product}>
            <SanitizedHTML
              content={description}
              allowedTags={allowedTags}
              allowedAttributes={allowedAttributes}
            />
          </GradientCollapse>
        ) : (
          <SanitizedHTML
            content={description}
            allowedTags={allowedTags}
            allowedAttributes={allowedAttributes}
          />
        )}
      </div>
    </div>
  )
}

const MemoizedProductDescription: MemoExoticComponent<
  typeof ProductDescription
> & { schema?: Record<string, string> } = memo(ProductDescription)

MemoizedProductDescription.schema = {
  title: 'admin/editor.product-description.title',
}

export default MemoizedProductDescription
