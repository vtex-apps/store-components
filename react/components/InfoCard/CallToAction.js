import React from 'react'
import { Button } from 'vtex.styleguide'
import { Link } from 'vtex.render-runtime'
import { useCssHandles } from 'vtex.css-handles'

import { callActionValues } from './SchemaTypes'

const CSS_HANDLES = ['infoCardCallActionContainer', 'infoCardCallActionText']

// eslint-disable-next-line @typescript-eslint/no-empty-function
const noop = () => {}

const CallToAction = ({ mode, text, url, linkTarget }) => {
  const { handles } = useCssHandles(CSS_HANDLES)

  if (mode === callActionValues.NONE) {
    return null
  }

  // eslint-disable-next-line no-shadow
  const ActionWrapper = ({ text, mode }) => {
    if (mode === callActionValues.BUTTON) {
      return <Button onClick={noop}>{text}</Button>
    }

    // Mode is link
    return (
      <p
        className={`${handles.infoCardCallActionText} link t-body b underline c-action-primary`}
      >
        {text}
      </p>
    )
  }

  return (
    <Link
      className={`${handles.infoCardCallActionContainer} mt6 mb6`}
      target={linkTarget}
      to={url}
      alt={text}
    >
      <ActionWrapper text={text} mode={mode} />
    </Link>
  )
}

export default CallToAction
