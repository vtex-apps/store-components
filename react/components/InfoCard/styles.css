.infoCardImageSizing {
  width: auto;
  height: auto;
}
