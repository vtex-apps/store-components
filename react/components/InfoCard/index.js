import classNames from 'classnames'
import PropTypes, { bool, string, oneOf } from 'prop-types'
import React, { memo } from 'react'
import { values } from 'ramda'
import { injectIntl } from 'react-intl'
import {
  useRuntime,
  useExperimentalLazyImagesContext,
} from 'vtex.render-runtime'
import { formatIOMessage } from 'vtex.native-types'
import { useCssHandles } from 'vtex.css-handles'
import RichText from 'vtex.rich-text/index'

import CallToAction from './CallToAction'
import LinkWrapper from './LinkWrapper'
import {
  textPositionTypes,
  textAlignmentTypes,
  callToActionModeTypes,
  textPostionValues,
  textAlignmentValues,
  textModeTypes,
} from './SchemaTypes'
import { SanitizedHTML } from '../SanitizedHTML'

const ALLOWED_TAGS = ['p', 'span', 'a', 'div', 'br']
const ALLOWED_ATTRS = {
  a: ['class', 'href', 'title', 'target'],
  span: ['class'],
  p: ['class'],
  div: ['class'],
}

const justifyTokens = {
  [textPostionValues.LEFT]: 'justify-start',
  [textPostionValues.CENTER]: 'justify-center',
  [textPostionValues.RIGHT]: 'justify-end',
}

const alignTokens = {
  [textAlignmentValues.LEFT]: 'tl',
  [textAlignmentValues.CENTER]: 'tc',
  [textAlignmentValues.RIGHT]: 'tr',
}

const itemsTokens = {
  [textAlignmentValues.LEFT]: 'items-start',
  [textAlignmentValues.CENTER]: 'items-center',
  [textAlignmentValues.RIGHT]: 'items-end',
}

const flexOrderTokens = {
  [textPostionValues.LEFT]: 'flex-row',
  [textPostionValues.RIGHT]: 'flex-row-reverse',
}

const defaultValues = {
  textPosition: textPositionTypes.TEXT_POSITION_LEFT.value,
  textAlignment: textAlignmentTypes.TEXT_ALIGNMENT_LEFT.value,
}

const getEnumValues = enumObject => values(enumObject).map(({ value }) => value)
const getEnumNames = enumObject => values(enumObject).map(({ name }) => name)

const safelyGetToken = (tokenMap, valueWanted, propName) =>
  tokenMap[valueWanted] || defaultValues[propName]

const getImageUrl = (isMobile, imageUrl, mobileImageUrl) =>
  !!mobileImageUrl && isMobile ? mobileImageUrl : imageUrl

const CSS_HANDLES = [
  'infoCardImageLinkWrapper',
  'infoCardContainer',
  'infoCardTextContainer',
  'infoCardHeadline',
  'infoCardSubhead',
  'infoCardBodyText',
  'infoCardImageContainer',
  'infoCardImage',
]

const InfoCard = ({
  isFullModeStyle,
  headline,
  subhead,
  bodyText,
  callToActionMode,
  callToActionText,
  callToActionUrl,
  callToActionLinkTarget,
  textPosition,
  textAlignment,
  imageUrl,
  mobileImageUrl,
  imageActionUrl,
  intl,
  htmlId,
  textMode,
  linkTarget,
  fetchpriority,
  preload,
}) => {
  const {
    hints: { mobile },
  } = useRuntime()

  const { lazyLoad } = useExperimentalLazyImagesContext()

  const { handles } = useCssHandles(CSS_HANDLES)
  const paddingClass =
    textPosition === textPostionValues.LEFT ? 'pr4-ns' : 'pl4-ns'

  // We ignore textAlignment tokens when full image mode
  const alignToken = isFullModeStyle
    ? safelyGetToken(alignTokens, textPosition, 'textPosition')
    : safelyGetToken(alignTokens, textAlignment, 'textAlignment')

  const itemsToken = isFullModeStyle
    ? safelyGetToken(itemsTokens, textPosition, 'textPosition')
    : safelyGetToken(itemsTokens, textAlignment, 'textAlignment')

  const justifyToken = safelyGetToken(
    justifyTokens,
    textPosition,
    'textPosition'
  )

  const flexOrderToken = safelyGetToken(
    flexOrderTokens,
    textPosition,
    'textPosition'
  )

  const finalImageUrl = getImageUrl(
    mobile,
    formatIOMessage({ id: imageUrl, intl }),
    formatIOMessage({ id: mobileImageUrl, intl })
  )

  const containerStyle = isFullModeStyle
    ? {
        /* If lazyloaded, the background image comes from the `data-bg` attribute
         * below. Otherwise, sets it here as background-image */
        ...(!lazyLoad && { backgroundImage: `url(${finalImageUrl})` }),
        backgroundSize: 'cover',
      }
    : {}

  const containerAttributes =
    isFullModeStyle && lazyLoad ? { 'data-bg': finalImageUrl } : {}

  const containerClasses = classNames(
    `${handles.infoCardContainer} items-center`,
    {
      [`flex-ns ${flexOrderToken} bg-base ph2-ns pb2 justify-between`]:
        !isFullModeStyle,
      [`bg-center bb b--muted-4 flex ${justifyToken}`]: isFullModeStyle,
      lazyload: lazyLoad,
    }
  )

  const linkWrapperClasses = `${handles.infoCardImageLinkWrapper} no-underline`

  const textContainerClasses = classNames(
    `${handles.infoCardTextContainer} flex flex-column mw-100`,
    {
      [`w-50-ns ph3-s ${itemsToken} ${paddingClass}`]: !isFullModeStyle,
      [`mh8-ns mh4-s w-40-ns ${itemsToken}`]: isFullModeStyle,
    }
  )

  const headlineClasses = `${handles.infoCardHeadline} t-heading-2 mt6 ${alignToken} c-on-base mw-100`

  const subheadClasses = `${handles.infoCardSubhead} t-body mt6 c-on-base ${alignToken} mw-100`

  const bodyTextClasses = `${handles.infoCardBodyText} t-body mt6 c-on-base ${alignToken} mw-100`

  return (
    <LinkWrapper
      imageActionUrl={formatIOMessage({ id: imageActionUrl, intl })}
      extraCondition={!isFullModeStyle}
      linkProps={{ className: linkWrapperClasses, target: linkTarget }}
    >
      <div
        className={containerClasses}
        style={containerStyle}
        data-testid="container"
        id={htmlId}
        {...containerAttributes}
      >
        <div className={textContainerClasses}>
          {headline &&
            (textMode === 'html' ? (
              <div className={headlineClasses}>
                <SanitizedHTML
                  content={formatIOMessage({ id: headline, intl })}
                  allowedTags={ALLOWED_TAGS}
                  allowedAttributes={ALLOWED_ATTRS}
                />
              </div>
            ) : (
              <RichText className={headlineClasses} text={headline} />
            ))}
          {subhead &&
            (textMode === 'html' ? (
              <div className={subheadClasses}>
                <SanitizedHTML
                  content={formatIOMessage({ id: subhead, intl })}
                  allowedTags={ALLOWED_TAGS}
                  allowedAttributes={ALLOWED_ATTRS}
                />
              </div>
            ) : (
              <RichText className={subheadClasses} text={subhead} />
            ))}
          {bodyText &&
            (textMode === 'html' ? (
              <div className={bodyTextClasses}>
                <SanitizedHTML
                  content={formatIOMessage({ id: bodyText, intl })}
                  allowedTags={ALLOWED_TAGS}
                  allowedAttributes={ALLOWED_ATTRS}
                />
              </div>
            ) : (
              <RichText className={bodyTextClasses} text={bodyText} />
            ))}
          <CallToAction
            mode={callToActionMode}
            text={formatIOMessage({ id: callToActionText, intl })}
            url={formatIOMessage({ id: callToActionUrl, intl })}
            linkTarget={callToActionLinkTarget}
          />
        </div>
        {!isFullModeStyle && (
          <div className={`${handles.infoCardImageContainer} w-50-ns`}>
            <LinkWrapper
              imageActionUrl={formatIOMessage({ id: imageActionUrl, intl })}
              linkProps={{ target: linkTarget }}
            >
              <img
                className={handles.infoCardImage}
                src={finalImageUrl}
                style={{ objectFit: 'cover' }}
                alt={formatIOMessage({ id: callToActionText, intl })}
                data-testid="half-image"
                fetchPriority={fetchpriority}
                {...(preload
                  ? {
                      'data-vtex-preload': 'true',
                    }
                  : {})}
              />
            </LinkWrapper>
          </div>
        )}
      </div>
    </LinkWrapper>
  )
}

const MemoizedInfoCard = memo(injectIntl(InfoCard))

MemoizedInfoCard.propTypes = {
  blockClass: string,
  isFullModeStyle: bool,
  textPosition: oneOf(getEnumValues(textPositionTypes)),
  headline: string,
  subhead: string,
  bodyText: string,
  callToActionMode: oneOf(getEnumValues(callToActionModeTypes)),
  callToActionText: string,
  callToActionUrl: string,
  imageUrl: string,
  mobileImageUrl: string,
  textAlignment: oneOf(getEnumValues(textAlignmentTypes)),
  imageActionUrl: string,
  intl: PropTypes.object,
  htmlId: string,
  textMode: oneOf(getEnumValues(textModeTypes)),
  // https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a
  linkTarget: oneOf(['_self', '_blank', '_parent', '_top']),
  callToActionLinkTarget: oneOf(['_self', '_blank', '_parent', '_top']),
}

MemoizedInfoCard.defaultProps = {
  isFullModeStyle: false,
  textPosition: textPositionTypes.TEXT_POSITION_LEFT.value,
  headline: '',
  subhead: '',
  bodyText: '',
  callToActionMode: callToActionModeTypes.CALL_ACTION_BUTTON.value,
  callToActionText: '',
  callToActionUrl: '',
  imageUrl: '',
  mobileImageUrl: '',
  textAlignment: textAlignmentTypes.TEXT_ALIGNMENT_LEFT.value,
  textMode: textModeTypes.TEXT_MODE_HTML.value,
  linkTarget: '_self',
  callToActionLinkTarget: '_self',
  fetchpriority: 'auto',
  preload: false,
}

MemoizedInfoCard.schema = {
  title: 'admin/editor.info-card.title',
  description: 'admin/editor.info-card.description',
  type: 'object',
  properties: {
    isFullModeStyle: {
      title: 'admin/editor.info-card.isFullModeStyle.title',
      description: 'admin/editor.info-card.isFullModeStyle.description',
      type: 'boolean',
      default: false,
      isLayout: true,
    },
    textPosition: {
      title: 'admin/editor.info-card.textPosition.title',
      description: 'admin/editor.info-card.textPosition.description',
      type: 'string',
      enum: getEnumValues(textPositionTypes),
      enumNames: getEnumNames(textPositionTypes),
      default: textPositionTypes.TEXT_POSITION_LEFT.value,
      isLayout: true,
    },
    callToActionMode: {
      title: 'admin/editor.info-card.callToActionMode.title',
      description: 'admin/editor.info-card.callToActionMode.description',
      type: 'string',
      enum: getEnumValues(callToActionModeTypes),
      enumNames: getEnumNames(callToActionModeTypes),
      default: callToActionModeTypes.CALL_ACTION_BUTTON.value,
      isLayout: true,
    },
    textAlignment: {
      title: 'admin/editor.info-card.textAlignment.title',
      description: 'admin/editor.info-card.textAlignment.description',
      type: 'string',
      default: textAlignmentTypes.TEXT_ALIGNMENT_LEFT.value,
      enum: getEnumValues(textAlignmentTypes),
      enumNames: getEnumNames(textAlignmentTypes),
      isLayout: true,
    },
    textMode: {
      title: 'admin/editor.info-card.textMode.title',
      description: 'admin/editor.info-card.textMode.description',
      type: 'string',
      default: textModeTypes.TEXT_MODE_HTML.value,
      enum: getEnumValues(textModeTypes),
      enumNames: getEnumNames(textModeTypes),
      isLayout: true,
    },
    blockClass: {
      title: 'admin/editor.blockClass.title',
      description: 'admin/editor.blockClass.description',
      type: 'string',
      isLayout: true,
    },
  },
}

export default MemoizedInfoCard
