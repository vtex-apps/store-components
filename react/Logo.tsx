import React from 'react'
import { Link, useRuntime } from 'vtex.render-runtime'
import type { RenderContext } from 'vtex.render-runtime'
import { useCssHandles } from 'vtex.css-handles'
import type { CssHandlesTypes } from 'vtex.css-handles'
import * as Amp from 'react-amphtml'

import Placeholder from './components/Logo/LogoPlaceholder'
import styles from './components/Logo/styles.css'

const CSS_HANDLES = ['logoLink', 'logoImage', 'logoContainer'] as const

interface Props {
  /** Logo's URL */
  url: string
  /** Title to be displayed as alt text */
  title: string
  /** Logo's width */
  width?: string | number
  /** Logo's height */
  height?: string | number
  /** Link to redirect users on click */
  href?: string
  /** Logo's width on mobile */
  mobileWidth?: string | number
  /** Logo's height on mobile */
  mobileHeight?: string | number
  /** Used to override default CSS handles */
  classes?: CssHandlesTypes.CustomClasses<typeof CSS_HANDLES>
}

/**
 * Store's logo.
 */
function Logo({
  url,
  href,
  width,
  height,
  title,
  mobileWidth,
  mobileHeight,
  classes,
}: Props) {
  const {
    amp,
    account,
    hints: { mobile },
  } = useRuntime() as RenderContext

  const { handles } = useCssHandles(CSS_HANDLES, { classes })
  const logoClassNames = `store-logo ${handles.logoContainer} ${
    mobile ? styles.sizeMobile : styles.sizeDesktop
  }`

  const imgWidth = mobile && mobileWidth ? mobileWidth : width
  const imgHeight = mobile && mobileHeight ? mobileHeight : height

  const imageUrl = url?.replace(/{{account}}/g, account)

  let image = null

  if (amp && url) {
    image = (
      <Amp.AmpImg
        specName="default"
        width={imgWidth}
        height={imgHeight}
        alt={title}
        src={imageUrl}
        className={handles.logoImage}
      />
    )
  } else if (url) {
    image = (
      <img
        src={imageUrl}
        width={imgWidth}
        height={imgHeight}
        alt={title}
        className={handles.logoImage}
      />
    )
  }

  const logo = (
    <span className={`${logoClassNames} pv4 ph6`}>
      {url ? (
        image
      ) : (
        <Placeholder width={width} height={height} title={title} />
      )}
    </span>
  )

  return href ? (
    <Link to={href} className={handles.logoLink}>
      {logo}
    </Link>
  ) : (
    logo
  )
}

Logo.schema = {
  title: 'admin/editor.logo.title',
  description: 'admin/editor.logo.description',
  type: 'object',
  properties: {
    href: {
      title: 'admin/editor.logo.href.title',
      description: 'admin/editor.logo.href.description',
      type: 'string',
    },
    url: {
      title: 'admin/editor.logo.url.title',
      description: 'admin/editor.logo.url.description',
      type: 'string',
      widget: {
        'ui:widget': 'image-uploader',
      },
    },
    width: {
      title: 'admin/editor.logo.width.title',
      description: 'admin/editor.logo.width.description',
      type: 'number',
      isLayout: true,
    },
    height: {
      title: 'admin/editor.logo.height.title',
      description: 'admin/editor.logo.height.description',
      type: 'number',
      isLayout: true,
    },
    title: {
      title: 'admin/editor.logo.title.title',
      description: 'admin/editor.logo.title.description',
      type: 'string',
    },
    mobileWidth: {
      title: 'admin/editor.logo.mobileWidth.title',
      description: 'admin/editor.logo.mobileWidth.description',
      type: 'number',
      isLayout: true,
    },
    mobileHeight: {
      title: 'admin/editor.logo.mobileHeight.title',
      description: 'admin/editor.logo.mobileHeight.description',
      type: 'number',
      isLayout: true,
    },
  },
}

export default Logo
