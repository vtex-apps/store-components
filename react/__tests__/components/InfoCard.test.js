import React from 'react'
import { render } from '@vtex/test-tools/react'

import InfoCard from '../../InfoCard'

describe('<InfoCard />', () => {
  const renderComponent = props => {
    const defaultProps = {
      isFullModeStyle: false,
      headline: 'MY HEADLINE',
      subhead: 'MY SUBHEAD',
      imageUrl: 'my-image.com/image.png',
      callToActionText: 'CLICK HERE',
      callToActionUrl: 'classic-shoes/p',
    }

    return render(<InfoCard {...defaultProps} {...props} />)
  }

  it('should render with full mode style', () => {
    const { asFragment, getByTestId } = renderComponent({
      isFullModeStyle: true,
    })

    expect(asFragment()).toBeDefined()
    const container = getByTestId('container')

    expect(container.style['background-image']).toBeDefined()
    expect(asFragment()).toMatchSnapshot()
  })
  it('should render with image and text side by side', () => {
    const { asFragment, getByTestId } = renderComponent()

    expect(asFragment()).toBeDefined()
    const halfImage = getByTestId('half-image')

    expect(halfImage).toBeDefined()
    expect(asFragment()).toMatchSnapshot()
  })
  it('should render without a call to action', () => {
    const { asFragment } = renderComponent({
      isFullModeStyle: true,
      callToActionMode: 'none',
    })

    expect(asFragment()).toBeDefined()
    expect(asFragment()).toMatchSnapshot()
  })
  it('should wrap half image in a link with imageActionUrl', () => {
    const { asFragment } = renderComponent({
      isFullModeStyle: false,
      callToActionMode: 'none',
      imageActionUrl: 'classic-shoes/p',
    })

    expect(asFragment()).toBeDefined()
    expect(asFragment()).toMatchSnapshot()
  })
  it('should wrap whole container in a link with imageActionUrl', () => {
    const { asFragment } = renderComponent({
      isFullModeStyle: true,
      callToActionMode: 'none',
      imageActionUrl: 'classic-shoes/p',
      textAlignment: 'right',
      textPosition: 'center',
    })

    expect(asFragment()).toBeDefined()
    expect(asFragment()).toMatchSnapshot()
  })
  it('should insert span with class on headline', () => {
    const { asFragment } = renderComponent({
      isFullModeStyle: true,
      callToActionMode: 'none',
      textAlignment: 'right',
      textPosition: 'center',
      headline:
        'HEADLINE <span class="my-custom-class">THIS IS BOOOLD AND BLUE</span> HEADLINE STILL',
    })

    expect(asFragment()).toBeDefined()
    expect(asFragment()).toMatchSnapshot()
  })
  it('should not render subhead', () => {
    const { asFragment } = renderComponent({
      isFullModeStyle: true,
      callToActionMode: 'none',
      textAlignment: 'right',
      textPosition: 'center',
      headline:
        'HEADLINE <span class="my-custom-class">THIS IS BOOOLD AND BLUE</span> HEADLINE STILL',
      subhead: '',
    })

    expect(asFragment()).toBeDefined()
    expect(asFragment()).toMatchSnapshot()
  })
  it("should render subhead and headline using the RichText component when textMode is set to 'rich-text'", () => {
    const { asFragment } = renderComponent({
      isFullModeStyle: true,
      callToActionMode: 'none',
      textAlignment: 'right',
      textPosition: 'center',
      textMode: 'rich-text',
      headline: 'This is a headline, and should be inside a rich-text.',
      subhead: 'This is a subhead, and should be inside a rich-text.',
    })

    expect(asFragment()).toBeDefined()
    expect(asFragment()).toMatchSnapshot()
  })
  it('should not render bodyText', () => {
    const { asFragment } = renderComponent({
      isFullModeStyle: true,
      callToActionMode: 'none',
      textAlignment: 'right',
      textPosition: 'center',
      headline:
        'HEADLINE <span class="my-custom-class">THIS IS BOOOLD AND BLUE</span> HEADLINE STILL',
      subhead: '',
      bodyText: '',
    })

    expect(asFragment()).toBeDefined()
    expect(asFragment()).toMatchSnapshot()
  })
  it("should render bodyText, subhead and headline using the RichText component when textMode is set to 'rich-text'", () => {
    const { asFragment } = renderComponent({
      isFullModeStyle: true,
      callToActionMode: 'none',
      textAlignment: 'right',
      textPosition: 'center',
      textMode: 'rich-text',
      headline: 'This is a headline, and should be inside a rich-text.',
      subhead: 'This is a subhead, and should be inside a rich-text.',
      bodyText: 'This is a bodyText, and should be inside a rich-text.',
    })

    expect(asFragment()).toBeDefined()
    expect(asFragment()).toMatchSnapshot()
  })
})
