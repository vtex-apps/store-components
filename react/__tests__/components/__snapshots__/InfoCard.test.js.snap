// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoCard /> should insert span with class on headline 1`] = `
<DocumentFragment>
  <div
    class="infoCardContainer items-center bg-center bb b--muted-4 flex justify-center"
    data-testid="container"
    style="background-image: url(my-image.com/image.png); background-size: cover;"
  >
    <div
      class="infoCardTextContainer flex flex-column mw-100 mh8-ns mh4-s w-40-ns items-center"
    >
      <div
        class="infoCardHeadline t-heading-2 mt6 tc c-on-base mw-100"
      >
        <div
          style="display: contents;"
        >
          HEADLINE 
          <span
            class="my-custom-class"
          >
            THIS IS BOOOLD AND BLUE
          </span>
           HEADLINE STILL
        </div>
      </div>
      <div
        class="infoCardSubhead t-body mt6 c-on-base tc mw-100"
      >
        <div
          style="display: contents;"
        >
          MY SUBHEAD
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<InfoCard /> should not render bodyText 1`] = `
<DocumentFragment>
  <div
    class="infoCardContainer items-center bg-center bb b--muted-4 flex justify-center"
    data-testid="container"
    style="background-image: url(my-image.com/image.png); background-size: cover;"
  >
    <div
      class="infoCardTextContainer flex flex-column mw-100 mh8-ns mh4-s w-40-ns items-center"
    >
      <div
        class="infoCardHeadline t-heading-2 mt6 tc c-on-base mw-100"
      >
        <div
          style="display: contents;"
        >
          HEADLINE 
          <span
            class="my-custom-class"
          >
            THIS IS BOOOLD AND BLUE
          </span>
           HEADLINE STILL
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<InfoCard /> should not render subhead 1`] = `
<DocumentFragment>
  <div
    class="infoCardContainer items-center bg-center bb b--muted-4 flex justify-center"
    data-testid="container"
    style="background-image: url(my-image.com/image.png); background-size: cover;"
  >
    <div
      class="infoCardTextContainer flex flex-column mw-100 mh8-ns mh4-s w-40-ns items-center"
    >
      <div
        class="infoCardHeadline t-heading-2 mt6 tc c-on-base mw-100"
      >
        <div
          style="display: contents;"
        >
          HEADLINE 
          <span
            class="my-custom-class"
          >
            THIS IS BOOOLD AND BLUE
          </span>
           HEADLINE STILL
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<InfoCard /> should render bodyText, subhead and headline using the RichText component when textMode is set to 'rich-text' 1`] = `
<DocumentFragment>
  <div
    class="infoCardContainer items-center bg-center bb b--muted-4 flex justify-center"
    data-testid="container"
    style="background-image: url(my-image.com/image.png); background-size: cover;"
  >
    <div
      class="infoCardTextContainer flex flex-column mw-100 mh8-ns mh4-s w-40-ns items-center"
    >
      <div
        data-testid="rich-text"
      >
        This is a headline, and should be inside a rich-text.
      </div>
      <div
        data-testid="rich-text"
      >
        This is a subhead, and should be inside a rich-text.
      </div>
      <div
        data-testid="rich-text"
      >
        This is a bodyText, and should be inside a rich-text.
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<InfoCard /> should render subhead and headline using the RichText component when textMode is set to 'rich-text' 1`] = `
<DocumentFragment>
  <div
    class="infoCardContainer items-center bg-center bb b--muted-4 flex justify-center"
    data-testid="container"
    style="background-image: url(my-image.com/image.png); background-size: cover;"
  >
    <div
      class="infoCardTextContainer flex flex-column mw-100 mh8-ns mh4-s w-40-ns items-center"
    >
      <div
        data-testid="rich-text"
      >
        This is a headline, and should be inside a rich-text.
      </div>
      <div
        data-testid="rich-text"
      >
        This is a subhead, and should be inside a rich-text.
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<InfoCard /> should render with full mode style 1`] = `
<DocumentFragment>
  <div
    class="infoCardContainer items-center bg-center bb b--muted-4 flex justify-start"
    data-testid="container"
    style="background-image: url(my-image.com/image.png); background-size: cover;"
  >
    <div
      class="infoCardTextContainer flex flex-column mw-100 mh8-ns mh4-s w-40-ns items-start"
    >
      <div
        class="infoCardHeadline t-heading-2 mt6 tl c-on-base mw-100"
      >
        <div
          style="display: contents;"
        >
          MY HEADLINE
        </div>
      </div>
      <div
        class="infoCardSubhead t-body mt6 c-on-base tl mw-100"
      >
        <div
          style="display: contents;"
        >
          MY SUBHEAD
        </div>
      </div>
      <a
        href="dummy"
      >
        <button>
          CLICK HERE
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<InfoCard /> should render with image and text side by side 1`] = `
<DocumentFragment>
  <div
    class="infoCardContainer items-center flex-ns flex-row bg-base ph2-ns pb2 justify-between"
    data-testid="container"
  >
    <div
      class="infoCardTextContainer flex flex-column mw-100 w-50-ns ph3-s items-start pr4-ns"
    >
      <div
        class="infoCardHeadline t-heading-2 mt6 tl c-on-base mw-100"
      >
        <div
          style="display: contents;"
        >
          MY HEADLINE
        </div>
      </div>
      <div
        class="infoCardSubhead t-body mt6 c-on-base tl mw-100"
      >
        <div
          style="display: contents;"
        >
          MY SUBHEAD
        </div>
      </div>
      <a
        href="dummy"
      >
        <button>
          CLICK HERE
        </button>
      </a>
    </div>
    <div
      class="infoCardImageContainer w-50-ns"
    >
      <img
        alt="CLICK HERE"
        class="infoCardImage"
        data-testid="half-image"
        src="my-image.com/image.png"
        style="object-fit: cover;"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`<InfoCard /> should render without a call to action 1`] = `
<DocumentFragment>
  <div
    class="infoCardContainer items-center bg-center bb b--muted-4 flex justify-start"
    data-testid="container"
    style="background-image: url(my-image.com/image.png); background-size: cover;"
  >
    <div
      class="infoCardTextContainer flex flex-column mw-100 mh8-ns mh4-s w-40-ns items-start"
    >
      <div
        class="infoCardHeadline t-heading-2 mt6 tl c-on-base mw-100"
      >
        <div
          style="display: contents;"
        >
          MY HEADLINE
        </div>
      </div>
      <div
        class="infoCardSubhead t-body mt6 c-on-base tl mw-100"
      >
        <div
          style="display: contents;"
        >
          MY SUBHEAD
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<InfoCard /> should wrap half image in a link with imageActionUrl 1`] = `
<DocumentFragment>
  <div
    class="infoCardContainer items-center flex-ns flex-row bg-base ph2-ns pb2 justify-between"
    data-testid="container"
  >
    <div
      class="infoCardTextContainer flex flex-column mw-100 w-50-ns ph3-s items-start pr4-ns"
    >
      <div
        class="infoCardHeadline t-heading-2 mt6 tl c-on-base mw-100"
      >
        <div
          style="display: contents;"
        >
          MY HEADLINE
        </div>
      </div>
      <div
        class="infoCardSubhead t-body mt6 c-on-base tl mw-100"
      >
        <div
          style="display: contents;"
        >
          MY SUBHEAD
        </div>
      </div>
    </div>
    <div
      class="infoCardImageContainer w-50-ns"
    >
      <a
        href="dummy"
      >
        <img
          alt="CLICK HERE"
          class="infoCardImage"
          data-testid="half-image"
          src="my-image.com/image.png"
          style="object-fit: cover;"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<InfoCard /> should wrap whole container in a link with imageActionUrl 1`] = `
<DocumentFragment>
  <a
    href="dummy"
  >
    <div
      class="infoCardContainer items-center bg-center bb b--muted-4 flex justify-center"
      data-testid="container"
      style="background-image: url(my-image.com/image.png); background-size: cover;"
    >
      <div
        class="infoCardTextContainer flex flex-column mw-100 mh8-ns mh4-s w-40-ns items-center"
      >
        <div
          class="infoCardHeadline t-heading-2 mt6 tc c-on-base mw-100"
        >
          <div
            style="display: contents;"
          >
            MY HEADLINE
          </div>
        </div>
        <div
          class="infoCardSubhead t-body mt6 c-on-base tc mw-100"
        >
          <div
            style="display: contents;"
          >
            MY SUBHEAD
          </div>
        </div>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
